src/aws-cpp-sdk-core/include/smithy/client/AwsSmithyClientBase.h

```diff
@@ -119,10 +123,76 @@ namespace client
             m_userAgent = Aws::Client::ComputeUserAgentString(m_clientConfig.get());
         }
 
-        AwsSmithyClientBase(const AwsSmithyClientBase&) = delete;
-        AwsSmithyClientBase(AwsSmithyClientBase&&) = delete;
-        AwsSmithyClientBase& operator=(const AwsSmithyClientBase&) = delete;
-        AwsSmithyClientBase& operator=(AwsSmithyClientBase&&) = delete;
+
+        AwsSmithyClientBase& operator=(const AwsSmithyClientBase& target)
+        {
+            if (this != &target)
+            {
+                m_clientConfig = target.m_clientConfig;
+                m_serviceName = target.m_serviceName;
+                m_userAgent = target.m_userAgent;
+                m_httpClient = Aws::Http::CreateHttpClient(*target.m_clientConfig);
+
+                // marshaller could be of different types
+                if (auto jsonMarshaller = std::dynamic_pointer_cast<Aws::Client::JsonErrorMarshaller>(m_errorMarshaller)) {
```

Comment:
not really a fan of the the use of `dynamic_pointer_cast` here, I think this is a symptom of the fact that we shouldnt be declaring a custom copy constructor here, but we should be in `AwsSmithyClient.h`, because at that point in time we know what that type of error marshaller exists.

Also this will break the cmake arg [ENABLE_RTTI](https://github.com/aws/aws-sdk-cpp/blob/a94edca824ecbe8984a70e3e67707bb77d9580c8/CMakeLists.txt#L40) if we turn it to false, which some of our customers do.

The custom copy constructor and assignment operator can call down to `AwsSmithyClientBase` functions that take these things as arguments.